pkg/grpc/server_test.go

```diff
@@ -62,6 +63,42 @@ func (a *APIServerSuite) TestEnvValues() {
 	}
 }
 
+func waitForPortToBeFree(t *testing.T, port uint64) {
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+
+	timer := time.NewTimer(10 * time.Second)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			listener, err := net.Listen("tcp4", fmt.Sprintf(":%d", port))
+			if err != nil {
+				t.Logf("Port %d still in use on tcp4", port)
+				continue
+			}
+			if errClose := listener.Close(); errClose != nil {
+				t.Logf("Closing tcp4 listener on port %d failed: %v", port, errClose)
+			}
+
+			listener, err = net.Listen("tcp6", fmt.Sprintf(":%d", port))
+			if err != nil {
+				t.Logf("Port %d still in use on tcp6", port)
+				continue
+			}
+			if errClose := listener.Close(); errClose != nil {
+				t.Logf("Closing tcp6 listener on port %d failed: %v", port, errClose)
+			}
+
+			return
```

Comment:
I was thinking of returning an error here and assert in the test `require.NoError(waitForPortToBeFree(a.T(), testPort))`

Reply:
I didn't want that test state to depend on the "wait" helper. tbh. For us, it is essential that we have a good state after calling the wait function. If we have a few issues with binding/closing, it is not so crucial for us.

But I agree that it could be helpful to have logs if we have failures.